lib/devise/encryptable/encryptors/pbkdf2.rb

```diff
@@ -0,0 +1,25 @@
+begin
+  module Devise
+    module Encryptable
+      module Encryptors
+        class Pbkdf2 < Base
+          def self.compare(encrypted_password, password, stretches, salt, pepper)
+            value_to_test = self.digest(password, stretches, salt, pepper)
+            ActiveSupport::SecurityUtils.fixed_length_secure_compare(encrypted_password, value_to_test)
```

Comment:
If comparing unmatching values of differing length, this fails with mismatch length error. This should use `Devise.secure_compare` instead, which should also do a fixed length comparison, but also does an initial byte size/length check. 


```suggestion
            Devise.secure_compare(encrypted_password, value_to_test)
```